### PR TITLE
fix: restore scalar filter params in query hooks

### PR DESCRIPTION
**Pull request by kilbot**
## Summary
- unwrap simple Mango `$eq` selectors when reading query selectors
- serialize Orders and Coupons filter params as scalars instead of `[object Object]`
- stop removable `ButtonPill` clicks from bubbling into the surrounding trigger
- add regression coverage for Orders/Coupons filter serialization and removable pill behavior

## Testing
- `node --experimental-vm-modules /Users/kilbot/Projects/monorepo-v2/node_modules/.bin/jest tests/hook-filters.test.ts tests/query-state.test.ts --runInBand`
- `jest packages/components/src/button/button-pill.test.tsx --runInBand --silent`
- `pnpm --dir packages/query run lint`
- `pnpm --dir packages/query run typecheck`
- `pnpm --dir packages/components run lint`
- `pnpm --dir packages/components run typecheck`

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented remove-button clicks from triggering surrounding element interactions.
  * Query filtering now accepts and correctly handles "$eq"-style selectors and returns unwrapped selector values.

* **UI Improvements**
  * More reliable pill behavior: explicit active/loading states and improved remove handling for cashier and customer pills.
  * Normalized customer ID handling to avoid empty/null string edge cases.

* **Tests**
  * Added tests for button removal behavior, query filtering/unwrapping, selector normalization, and related guard logic.

### CODE REVIEW

File: packages/query/src/hooks/orders.tsx [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+	return value;
+};
+
 /**
  *
  */
 const filterApiQueryParams = (params: Record<string, any>) => {
+	for (const [key, value] of Object.entries(params)) {
+		params[key] = unwrapEqSelector(value);
+	}
```
[the remark this answers is not in the document]
Comment (kilbot) [reply packages/query/src/hooks/orders.tsx:9]: Fixed in 68da5b25f. `customer_id` is now renamed by key presence instead of truthiness, and `packages/query/tests/hook-filters.test.ts` now covers the guest case (`{ $eq: 0 } -> customer: 0`).

File: packages/core/src/screens/main/orders/filter-bar.tsx [hunk 2]
```diff
@@ -26,9 +26,16 @@ export function FilterBar({
...
+	const rawCustomerID = useObservableEagerState(
 		query.rxQuery$.pipe(map(() => query.getSelector('customer_id')))
 	);
+	const customerID = React.useMemo(() => {
+		if (rawCustomerID === null || rawCustomerID === undefined || rawCustomerID === '') {
+			return undefined;
+		}
+
+		return toNumber(rawCustomerID as string | number);
+	}, [rawCustomerID]);
```
[the remark this answers is not in the document]
Comment (kilbot) [reply packages/core/src/screens/main/orders/filter-bar.tsx]: Fixed in 68da5b25f. I moved customer-ID normalization into `normalizeSelectedCustomerID(...)`, which returns `undefined` for non-finite coercions and preserves `0`. The new `customer-filter-utils.test.ts` covers the normalization cases directly.

File: packages/core/src/screens/main/reports/filter-bar.tsx [hunk 3]
```diff
@@ -28,9 +28,16 @@ import { useGuestCustomer } from '../hooks/use-guest-customer';
...
+	const rawCustomerID = useObservableEagerState(
+		query.rxQuery$.pipe(map(() => query.getSelector('customer_id')))
 	);
+	const customerID = React.useMemo(() => {
+		if (rawCustomerID === null || rawCustomerID === undefined || rawCustomerID === '') {
+			return undefined;
+		}
+
+		return toNumber(rawCustomerID as string | number);
+	}, [rawCustomerID]);
```
[the remark this answers is not in the document]
Comment (kilbot) [reply packages/core/src/screens/main/reports/filter-bar.tsx]: Fixed in 68da5b25f. Reports now uses the same shared `normalizeSelectedCustomerID(...)` helper as Orders, so invalid coercions collapse to `undefined` while guest `0` is preserved. Covered by `customer-filter-utils.test.ts`.

File: packages/core/src/screens/main/components/order/filter-bar/customer-pill-guards.test.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+describe('customer pill guards', () => {
+	it('uses the selected combobox customer item to seed the pill label immediately', () => {
+		const customerPill = readFileSync(join(__dirname, 'customer-pill.tsx'), 'utf8');
+
+		expect(customerPill).toContain(
+			'const [selectedCustomer, setSelectedCustomer] = React.useState'
+		);
+		expect(customerPill).toContain('setSelectedCustomer(option.item ?? null);');
+		expect(customerPill).toContain('if (selectedCustomer?.id === customerID)');
+		expect(customerPill).toContain('extractNameFromJSON(customer)');
```
[the remark this answers is not in the document]
Comment (kilbot) [reply packages/core/src/screens/main/components/order/filter-bar/customer-pill-guards.test.ts:13]: Addressed in 68da5b25f. I removed the brittle source-string test entirely and replaced it with executable coverage: `customer-filter-utils.test.ts` now covers the customer-id normalization/entity-resolution behavior directly, while the existing selected-entity tests cover the observable selection flow. There is no remaining `readFileSync(...).toContain(...)` assertion in this area.
